test/e2e/bandwidth_test.go

```diff
@@ -54,3 +59,81 @@ func BenchmarkBandwidthIntraNode(b *testing.B) {
 		b.ReportMetric(bandwidthNum, bandwidthUnit)
 	}, b)
 }
+
+// BenchmarkBandwidthServiceLocalAccess runs the benchmark of bandwidth between Pods on same node.
+func BenchmarkBandwidthServiceLocalAccess(b *testing.B) {
+	withPerfTestSetup(func(data *TestData) {
+		b.StartTimer()
+
+		svc, err := data.createService("perftest-b", iperfPort, iperfPort, map[string]string{"antrea-e2e": "perftest-b"})
+		if err != nil {
+			b.Fatalf("Error when creating perftest service: %v", err)
+		}
+		if err := data.createPodOnNode("perftest-a", masterNodeName(), perftoolImage, nil, nil, nil, nil); err != nil {
+			b.Fatalf("Error when create the perftest client Pod: %v", err)
+		}
+		if err := data.podWaitForRunning(defaultTimeout, "perftest-a", testNamespace); err != nil {
+			b.Fatalf("Error when waiting the perftest client Pod: %v", err)
+		}
+		if err := data.createPodOnNode("perftest-b", masterNodeName(), perftoolImage, nil, nil, nil, []v1.ContainerPort{{Protocol: v1.ProtocolTCP, ContainerPort: iperfPort}}); err != nil {
+			b.Fatalf("Error when creating the perftest server Pod: %v", err)
+		}
+		if err := data.podWaitForRunning(defaultTimeout, "perftest-b", testNamespace); err != nil {
+			b.Fatalf("Error when getting the perftest server Pod IP: %v", err)
+		}
+		stdout, stderr, err := data.runCommandFromPod(testNamespace, "perftest-a", perftoolContainerName, []string{"bash", "-c", fmt.Sprintf("iperf3 -c %s|grep sender|awk '{print $7,$8}'", svc.Spec.ClusterIP)})
+		if err != nil {
+			b.Fatalf("Error when running iperf3 client: %v", err)
+		}
+		stdout = strings.TrimSpace(stdout)
+		results := strings.Split(stdout, " ")
+		if len(results) != 2 {
+			b.Fatalf("Error when parsing iperf result: cannot parse output '%s', stderr '%s'", stdout, stderr)
+		}
+		// Disable default output.
+		b.ReportMetric(0, "ns/op")
+		bandwidthNum, _ := strconv.ParseFloat(results[0], 64)
+		bandwidthUnit := strings.TrimSpace(results[1])
+		b.ReportMetric(bandwidthNum, bandwidthUnit)
+	}, b)
+}
+
+// BenchmarkBandwidthServiceRemoteAccess runs the benchmark of bandwidth between Pods on same node.
+func BenchmarkBandwidthServiceRemoteAccess(b *testing.B) {
```

Comment:
I think BenchmarkBandwidthServiceLocalAccess and BenchmarkBandwidthServiceRemoteAccess can share most code by parameterizing the node of Pods.

Reply:
Good point, updated.